### ISSUE TITLE
Wrong selection in source pane

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#1935]
@peller 

(I'm using Mac/FF)
- New HTML File (hi-fi page)
- Add a Button and CheckBox
- Select Button, copy, paste after Checkbox
- Split view
- On canvas, click-select first Button. (correct widget is selected in source pane)
- On canvas, click-select second Button. 

Result=> First button is still selected. This is true even if I save, close, reopen. However, if I bring up SmartInput and modify label on second button, then source pane selection gets fixed.

**Comment by JonFerraiolo** [comment]
-> @jhpedemonte (so this doesn't fall through cracks)

**Comment by jhpedemonte** [comment]
Assigning to @doronrosenberg, but may require expertise of @peller.

**Comment by doronrosenberg** [comment]
Selection in source view is broken in general

**Comment by doronrosenberg** [comment]
@jhpedemonte @JonFerraiolo 

So I debug this into orion/textview/textView::setSelection, with valid offsets given from our code.  This seems to be an orion issue as far as I can tell.

**Comment by JonFerraiolo** [comment]
cc'ing @aerwin - I thought @aerwin fixed some of our source selection issues at one point recently.

**Comment by aerwin** [comment]
Yes, I did under #1435. And, it looks like my fix for that scenario works on FF on Mac.

**Comment by doronrosenberg** [comment]
@aerwin #1435 works for me in split view - clicking on a widget in the source panel highlights the widget in the source panel. However, if I only have the source view open, clicking on a widget doesn't highlight it in the source view and doesn't highlight it in the outline panel.

**Comment by aerwin** [comment]
Per scrum, does it make sense to lump #2225 on my list with this issue?

**Comment by JonFerraiolo** [comment]
> Per scrum, does it make sense to lump #2225 on my list with this issue?

Probably. Maybe you and @doronrosenberg can coordinate, perhaps reassiging #2225 to him.

**Comment by doronrosenberg** [comment]
@JonFerraiolo @jhpedemonte 

The general issue with selection to me seems to be that the Outline palette is built around the visual editor objects, so things go bad when we deal with the source view, which has the simpler model objects.

**Comment by peller** [comment]
see #750

**Comment by doronrosenberg** [comment]
So I got selection in source pane kinda working - I have to manually focus the source pane, which isn't very good of a solution

**Comment by JonFerraiolo** [comment]
-> @peller

**Comment by peller** [comment]
-> M8

**Comment by JonFerraiolo** [comment]
@peller @jhpedemonte - Just checked - problem still exists - we still select the wrong button in source pane

**Comment by peller** [comment]
-> M9
